### DISCUSSION TITLE
Subscribe to keyboard event from select

### DISCUSSION THREAD

**Discussion by HertZz98** [zauberzeug/nicegui#3216] [Q&A]
### Question

Hello everyone,

I am currently working on a NiceGUI project where I need to trigger a search function when the user presses the "Enter" key after typing a search query in a ui.select component. I attempted to achieve this by using the .on('keydown', handle_key) method as shown below:

```py
async def create_search_bar() -> None:
    _search_input = ui.select(
        label="Search (type your search query here and confirm with \"Enter\")",
        options=from_cache('search_history'),
        with_input=True,
        new_value_mode='add-unique',
        clearable=True
    ).on('keydown', handle_key).classes('w-2/5')

def handle_key(e: KeyEventArguments):
    global _search_input
    if e.key.enter:
        search()

async def search():
    # Logic for search
```
I also tried using the ui.keyboard component as follows:

```py
async def create_search_bar() -> None:
    _search_input = ui.select(
        label="Search (type your search query here and confirm with \"Enter\")",
        options=from_cache('search_history'),
        with_input=True,
        new_value_mode='add-unique',
        clearable=True
    ).classes('w-2/5')
    ui.keyboard(on_key=handle_key, active=True)

def handle_key(e: KeyEventArguments):
    print("Hi")
    if e.key.name == 'Enter':
        search()
```
However, neither of these approaches worked as intended.

My goal is to ensure that the search function is called upon pressing the "Enter" key. I also tried using the `on_change=search` property, but this approach led to an issue where the input field would select the closest matching option from the dropdown instead of allowing a new search query to be typed and submitted.

I would greatly appreciate any advice or solutions on how to achieve this functionality within NiceGUI.

Thank you in advance for your assistance.

**Comment by falkoschindler** [comment]
Hi @HertZz98,

When using the generic event subscription with `.on()`, the event handler is called with `GenericEventArguments` rather than `KeyEventArguments`. Therefore accessing the key code works a bit differently:
```py
def handle_key(e: events.GenericEventArguments):
    if e.args['key'] == 'Enter':
        print('Enter pressed:', e.args)

ui.select(label='Search', options=['A', 'B', 'C'], with_input=True, new_value_mode='add-unique', clearable=True) \
    .on('keydown', handle_key)
```
